<!--
 - tracker-single-cam.launch
 -
 - Launch file for pi_vision face tracking, with input from a single
 - stationary UVC (USB) camera.  The UVC camera is assumed to be 640x480
 - and located at /dev/video0.  It is configured to be fairly narrow
 - field-of-view (as is typical for desktop/laptop USB cameras).
 - (full-angle FOV is set to 0.7175 or about 41 degrees).
-->
<launch>
    <node pkg="usb_cam" type="usb_cam_node" name="camera">
      <param name="video_device" value="/dev/video0"/>
      <param name="pixel_format" value="yuyv"/>
      <!-- Image size and type -->
      <param name="width" value="640"/>
      <param name="height" value="480"/>
    </node>

    <arg name="auto_face_tracking" default="True" />
    <node pkg="pi_face_tracker"
          name="body_face_tracker"
          type="face_tracker.py" output="screen">

        <!--remap the topics if using multiple cameras -->
        <remap from="input_rgb_image" to="camera/image_raw" />
        <remap from="output_image" to="camera/pi_face_tracker/image" />

        <param name="auto_face_tracking" value="$(arg auto_face_tracking)" />

        <param name="camera_name" value="camera" />

        <rosparam>
          use_haar_only: False
          auto_min_features: True
          min_features: 50
          abs_min_features: 6
          good_feature_distance: 5
          add_feature_distance: 10
          show_text: True
          show_features: True
          camera_fov_x: 0.7175
          max_face_size: 0.28
          expand_roi: 1.02
          flip_image: False
          feature_type: 0 <!-- 0 = Good Features To Track, 1 = SURF -->
        </rosparam>

        <param name="cascade_frontal_alt"
               value="$(find pi_face_tracker)/params/haarcascade_frontalface_alt.xml" />
        <param name="cascade_frontal_alt2"
               value="$(find pi_face_tracker)/params/haarcascade_frontalface_alt2.xml" />
  </node>
</launch>
